crates/plugins/mediator-coordination/src/plugin.rs

```diff
@@ -86,42 +82,45 @@ impl Plugin for MediatorCoordinationPlugin {
     }
 
     fn routes(&self) -> Router {
```

Comment:
Check this example https://gist.github.com/rust-play/f079f88a8b79f4cbfb665f6420742780

Reply:
The self in routes refers to the same instance as in mount because i am referring to the same instance of MediatorCoordinationPlugin.

Reply:
That is not possible the two functions have no relation, infact they are not aware 

> The self in routes refers to the same instance as in mount because i am referring to the same instance of MediatorCoordinationPlugin.

That is not possible the two functions have no relation, infact they are not aware of each other, i mean check my example again you will see what i mean

Reply:
> Check this example https://gist.github.com/rust-play/f079f88a8b79f4cbfb665f6420742780

Hi @Christiantyemele. Do you realize in your example you are cloning the object? That's why you now have two instances. I would recommend to both of you to further check the difference between `self`, `&self`, and `&mut self`.

Here is an update of your minimal working example:

```rust
#[derive(Clone)]
#[allow(unused)]
struct Add {
    a: i32,
    b: i32
}

impl Add {
    fn add_first(&mut self, c: i32, d: i32) {
        self.a = c + d
    }
    
    fn give_from_add_first(&self) -> i32 {
        self.a
    }
}

fn main() {
    let mut add = Add {a: 4, b: 5};
    add.add_first(1, 2);
    println!("See, the logic holds with references: {}", add.give_from_add_first())
}
```

Reply:
:sweat_smile:

Reply:
Thanks

Reply:
My bad @ndefokou i was the one wrong :sweat_smile: